src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,209 @@
+
+            if (titleLength == IntPtr.Zero)
+            {
+                return string.Empty;
+            }
+
+            var title = new StringBuilder(titleLength.ToInt32() + 1);
+
+            User32.SendMessage(window, User32.WM_GETTEXT, (IntPtr)(title.Capacity), title);
+            return title.ToString();
+        }
+
+        public static IEnumerable<IntPtr> GetTopLevelWindows()
+        {
+            var topLevelWindows = new List<IntPtr>();
+
+            var enumFunc = new User32.WNDENUMPROC((hWnd, lParam) => {
+                topLevelWindows.Add(hWnd);
+                return true;
+            });
+            User32.EnumWindows(enumFunc, IntPtr.Zero);
+
+            return topLevelWindows;
+        }
+
+        public static void KillProcess(Process process)
+        {
+            try
+            {
+                if ((process != null) && (!process.HasExited))
+                {
+                    process.Kill();
+                }
+            }
+            catch (Win32Exception e)
+            {
+                IntegrationLog.Current.Warning("The caller attempted to kill a process which either could not be terminated or is already terminating.");
+                IntegrationLog.Current.WriteLine($"    {e}");
+            }
+            catch (InvalidOperationException e)
+            {
+                IntegrationLog.Current.Warning("The caller attempted to kill a process which either does not exist or has already been terminated.");
+                IntegrationLog.Current.WriteLine($"    {e}");
+            }
+        }
+
+        public static void KillProcess(string processName)
+        {
+            foreach (var process in Process.GetProcessesByName(processName))
+            {
+                KillProcess(processName);
+            }
+        }
+
+        public static string LocateFile(string fileName, string downloadUrl = "")
+        {
+            var filePath = Path.Combine(Environment.CurrentDirectory, fileName);
+
+            if (File.Exists(filePath))
+            {
+                return filePath;
+            }
+
+            filePath = Path.Combine(typeof(IntegrationHelper).Assembly.Location, fileName);
+
+            if (File.Exists(filePath))
+            {
+                return filePath;
+            }
+
+            if (!string.IsNullOrWhiteSpace(downloadUrl))
+            {
+                filePath = Path.Combine(Environment.CurrentDirectory, fileName);
+
+                using (var webClient = new WebClient())
+                {
+                    webClient.DownloadFile("https://live.sysinternals.com/procdump.exe", filePath);
+                }
+
+                return filePath;
+            }
+
+            throw new FileNotFoundException($"Unable to locate the specified file. (FileName: '{fileName}')", fileName);
+        }
+
+        public static void Sleep(int sleepDelay = DefaultSleepDelay)
```

Comment:
We need to remove this API.  The use of `Thread.Sleep` is a sign of a test that is going to fail / behave unpredictably under load.  The existing uses of `Thread.Sleep` in our code base are the number 1 source of instability in our test infrastructure.  Any tests which use this API should be changed to use definitive sources of synchronization such as `ManualResetEvent`

CC @davkean